### PR TITLE
test: add miniflare environment regression test

### PR DESCRIPTION
**Pull request by mattcosta7**
This was a failing test for the miniflare environment as discussed in #1834 

Now that related changes to interceptors and cookies have shipped, and we have upgraded versions this test is green, and provides a regression check on future changes

The original error was something like this:

> ❯ test/node/regressions/miniflare.node.test.ts  (2 tests | 2 failed) 61ms
   ❯ test/node/regressions/miniflare.node.test.ts > returns a mocked response to an "http.get" request
     → Failed to get the 'credentials' property on 'Request': the property is not implemented.
   ❯ test/node/regressions/miniflare.node.test.ts > returns a mocked response to an "http.request" request
     → Failed to get the 'credentials' property on 'Request': the property is not implemented.

### CODE REVIEW

File: test/node/regressions/miniflare.node.test.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
+/**
```
Comment (mattcosta7) [inline test/node/regressions/miniflare.node.test.ts:1]: this is a copy of other tests for now, but may get further scoped in the future, or w can write a small cloudflare handler to validate (which would be ideal).

For now, this just throws on the property access we expect it to, a least

**Comment by kettanaito** [comment]
Just tested this with #1845 and updated `@mswjs/interceptors` and the tests are passing:

```
 ✓ test/node/regressions/miniflare.node.test.ts (2)
   ✓ returns a mocked response to an "http.get" request
   ✓ returns a mocked response to an "http.request" request
```

🎉

**Comment by kettanaito** [comment]
Updated the branch. Waiting for the tests to pass...

_Edit:_ Looks like we pushed the merge commit at the exact same time, @mattcosta7! :D That's fun.

**Review by kettanaito — APPROVED** [review]

**Comment by mattcosta7** [comment]
> Updated the branch. Waiting for the tests to pass...
> 
> _Edit:_ Looks like we pushed the merge commit at the exact same time, @mattcosta7! :D That's fun.

oh nice!  I realized that I had a different version of the lock file checked in, so downgraded pnpm locally and repushed that too

**Comment by kettanaito** [comment]
@mattcosta7, you are once again, a true wizard. Thank you so much.

**Comment by kettanaito** [comment]
## Released: v2.0.5 🎉

This has been released in v2.0.5!

- 📄 [**Release notes**](https://github.com/mswjs/msw/releases/tag/v2.0.5)
- 📦 [npm package](https://www.npmjs.com/package/msw/v/2.0.5)

Make sure to always update to the latest version (`npm i msw@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.

**Pull request merged**